core/amber/src/test/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/BatchToTupleConverterSpec.scala

```diff
@@ -0,0 +1,68 @@
+package edu.uci.ics.amber.engine.architecture.messaginglayer
+
+import edu.uci.ics.amber.engine.architecture.worker.neo.WorkerInternalQueue
+import com.softwaremill.macwire.wire
+import edu.uci.ics.amber.engine.architecture.worker.neo.WorkerInternalQueue.{
+  EndMarker,
+  EndOfAllMarker,
+  InputTuple,
+  SenderChangeMarker
+}
+import edu.uci.ics.amber.engine.common.ambermessage.WorkerMessage.{DataFrame, EndOfUpstream}
+import edu.uci.ics.amber.engine.common.ambertag.neo.VirtualIdentity.NamedActorVirtualIdentity
+import edu.uci.ics.amber.engine.common.tuple.ITuple
+import org.scalamock.scalatest.MockFactory
+import org.scalatest.flatspec.AnyFlatSpec
+
+class BatchToTupleConverterSpec extends AnyFlatSpec with MockFactory {
+  private val mockInternalQueue = mock[WorkerInternalQueue]
+  private val fakeID = NamedActorVirtualIdentity("testReceiver")
+
+  "tuple producer" should "break batch into tuples and output" in {
+    val batchToTupleConverter = wire[BatchToTupleConverter]
+    val inputBatch = DataFrame(Array.fill(4)(ITuple(1, 2, 3, 5, "9.8", 7.6)))
```

Comment:
let's create a class MockDataProducer?? I think we will be needing it quite a lot